stargz/reader/reader.go

```diff
@@ -133,22 +134,28 @@ func (gr *reader) Cache(opts ...CacheOption) (err error) {
 		filter = cacheOpts.filter
 	}
 
-	return gr.cacheWithReader(root, r, filter, cacheOpts.cacheOpts...)
+	eg, egCtx := errgroup.WithContext(context.Background())
+	eg.Go(func() error {
+		return gr.cacheWithReader(egCtx, eg, semaphore.NewWeighted(4),
```

Comment:
> semaphore.NewWeighted(4)

Should this corresponds to GOMAXPROCS?

Reply:
Curious, is it really allowed to recursively call `eg.Go` from `eg.Go`?

Reply:
Thanks for reviewing this. 

> Should this correspond to GOMAXPROCS?

Fixed it on the latest push.

> Curious, is it really allowed to recursively call eg.Go from eg.Go?

According to https://github.com/golang/go/issues/27837#issuecomment-612163144, this seems an allowed pattern.